luigi/worker.py

```diff
@@ -1127,8 +1132,16 @@ def _keep_alive(self, get_work_response):
             return get_work_response.n_pending_last_scheduled > 0
         elif self._config.count_uniques:
             return get_work_response.n_unique_pending > 0
+        elif get_work_response.n_pending_tasks <= 0:
+            return False
+        elif not self._config.keep_alive_ttl:
+            return True
+        elif not self._idle_since:
+            return True
         else:
-            return get_work_response.n_pending_tasks > 0
+            time_to_shutdown = self._idle_since + self._config.keep_alive_ttl - datetime.datetime.now()
+            logger.debug("%s until shutdown", time_to_shutdown)
```

Comment:
should this log the worker id? is the worker_id accessible here? Just seems like a valuable thing to know within the logging.